### ISSUE TITLE
Faster R-CNN training in TensorFlow < 1.4 ?

### ISSUE THREAD

**Issue by csnemes2** [ppwwyyxx/tensorpack#578]
Can I retrain Faster R-CNN with TF 1.3?
Evaluation worked perfectly with TF1.3,  and I can confirm your published results.

BUT, when I train I run into an assert at https://github.com/ppwwyyxx/tensorpack/blob/master/tensorpack/models/batch_norm.py

> 
> if use_local_stat:
>        blabla
>else:
>        if ctx.is_training:
>            assert get_tf_version_number() >= 1.4, \
>                "Fine tuning a BatchNorm model with fixed statistics is only " \
>"supported after https://github.com/tensorflow/tensorflow/pull/12580 "

Without pain, Is there any chance to set use_local_stat to True?
According to my quick check

- Faster RCNN relies on BNReLUs (see https://github.com/ppwwyyxx/tensorpack/blob/master/tensorpack/models/nonlin.py)
- BNReLU  defaults use_local_stat to None
- I can modify this but still hesitating, would it work? Any suggestion?

**Comment by ppwwyyxx** [comment]
You cannot. That's why it requires TF>=1.4

**Comment by csnemes2** [comment]
Thanks for the quick and clear answer, I am already downloading the 1.4 :)
Any chance to update the readme? 
I have just checked it and no hints, just something tf>=1.2 which is true but not very tight :)

**Comment by sharpstill** [comment]
tf 1.5 has already release , which fully support dynamic graph( like in PyTorch), which will ease the burden of debug breakpoint tf. Will tensorpack supply a switch argument to open dynamic graph let us to easy debug?

**Comment by ppwwyyxx** [comment]
> Any chance to update the readme?

The readme is correct. Tensorpack depends on TF>=1.2; FasterRCNN depends on TF>=1.4. Both are mentioned in the corresponding readme.

@sharpstill I'll look at that in the future at #463.
